o2a/mappers/java_mapper.py

```diff
@@ -0,0 +1,115 @@
+        self.archive_extractor = ArchiveExtractor(oozie_node=oozie_node, props=self.props)
+        self.main_class: Optional[str] = None
+        self.java_opts: Optional[List[str]] = None
+        self.args: Optional[List[str]] = None
+        self.hdfs_files: Optional[List[str]] = None
+        self.hdfs_archives: Optional[List[str]] = None
+        self.prepare_extension: PrepareMapperExtension = PrepareMapperExtension(self)
+        self.jar_files: List[str] = jar_files if jar_files else []
+        self.jar_files_in_hdfs: List[str] = []
+        self._compute_jar_files_in_hdfs()
+
+    def on_parse_node(self):
+        super().on_parse_node()
+        _, self.hdfs_files = self.file_extractor.parse_node()
+        _, self.hdfs_archives = self.archive_extractor.parse_node()
+        self._extract_java_data()
+
+    def to_tasks_and_relations(self):
+        action_task = Task(
+            task_id=self.name,
+            template_name="java.tpl",
+            template_params=dict(
+                props=self.props,
+                hdfs_files=self.hdfs_files,
+                hdfs_archives=self.hdfs_archives,
+                main_class=self.main_class,
+                jar_files_in_hdfs=self.jar_files_in_hdfs,
+                args=self.args,
+            ),
+        )
+        tasks = [action_task]
+        relations: List[Relation] = []
+        prepare_task = self.prepare_extension.get_prepare_task()
+        if prepare_task:
+            tasks, relations = self.prepend_task(prepare_task, tasks, relations)
+        return tasks, relations
+
+    def required_imports(self) -> Set[str]:
+        return {"from airflow.utils import dates", "from airflow.contrib.operators import dataproc_operator"}
+
+    def _compute_jar_files_in_hdfs(self):
+        hdfs_app_prefix = self.props.job_properties["oozie.wf.application.path"]
+        for file in self.jar_files:
+            self.jar_files_in_hdfs.append(hdfs_app_prefix + "/" + LIB_FOLDER + "/" + file)
+
+    def _extract_java_data(self):
+        """Extracts Java node data."""
+        root = self.oozie_node
+        props = self.props
+        self.java_opts: List[str] = []
+        if "mapred.child.java.opts" in props.merged:
+            self.java_opts.extend(props.merged["mapred.child.java.opts"].split(" "))
+        if "mapreduce.map.java.opts" in props.merged:
+            self.java_opts.extend(props.merged["mapreduce.map.java.opts"].split(" "))
+        self.main_class = xml_utils.get_tag_el_text(root=root, tag="main-class", props=props)
+        java_opts_string = xml_utils.get_tag_el_text(root=root, tag="java-opts", props=props)
+        if java_opts_string:
+            self.java_opts.extend(java_opts_string.split(" "))
+        else:
+            java_opt_node_array = xml_utils.find_nodes_by_tag(root=root, tag="java-opt")
+            if java_opt_node_array:
+                for node in java_opt_node_array:
+                    self.java_opts.append(node.text)
```

Comment:
Is this node can not contain EL expression?

Reply:
Ah right. Will update.

Reply:
Fixed.